### PR TITLE
fix: respect the after recovery return to URL from config

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Related issue(s)
Fixes https://github.com/ory-corp/cloud/issues/1405

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: x/http_secure_redirect.go [hunk 1]
```diff
@@ -76,12 +76,14 @@ func SecureRedirectToIsAllowedHost(returnTo *url.URL, allowed url.URL) bool {
...
-func TakeOverReturnToParameter(from string, to string) (string, error) {
+// TakeOverReturnToParameter carries over the return_to parameter to a new URL
+// If `from` does not contain the `return_to` query parameter, the first non-empty value from `fallback` is used instead.
+func TakeOverReturnToParameter(from string, to string, fallback ...string) (string, error) {
 	fromURL, err := url.Parse(from)
 	if err != nil {
 		return "", err
 	}
-	returnTo := fromURL.Query().Get("return_to")
+	returnTo := stringsx.Coalesce(append([]string{fromURL.Query().Get("return_to")}, fallback...)...)
```
Comment (jonas-jonas) [inline x/http_secure_redirect.go:86]: Couldn't find a better way to combine the string value with a []string in a var-args parameter...

Comment (zepatrik) [reply x/http_secure_redirect.go:86]: I don't think there is a better one.

**Review by zepatrik — DISMISSED** [review]
Very nice :tada:

**Comment by jonas-jonas** [comment]
@zepatrik I realized, that we probably want the settings flow of the recovery flow behave the same way as the normal settings flow, if no `returnTo` was set in the config or the creation URL. With my code before, we would always return the user to the `defaultReturnTo` after submitting the settings, but the settings flow returns to the settings UI in that case. So I added that behavior here too. PTAL.

**Review by CaptainStandby — APPROVED** [review]
LGTM

**Review by Benehiko — APPROVED** [review]
LGTM :)

**Pull request merged**